### PR TITLE
feat: implement Typography component

### PR DESCRIPTION
**Pull request by baonguyenhsts**
## Summary
Create new component: Typography

## Check List
- [x] Your code builds clean without any errors or warnings.
- [x] Check coding convention.
- [x] Updated Storybook components and stories to reflect any changes made to the UI.
- [x] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness

Screenshot 2024-04-22 at 12 55 21 AM

### CODE REVIEW

**Review by ledutu2 — APPROVED** [review]

**Review by huydosgtech — APPROVED** [review]

**Review by loido — APPROVED** [review]

File: src/components/Typography/Typography.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ * @param {string} [props.color] - The color of the text.
+ * @param {...TextProps} [props.rest] - Additional text properties that can be passed down to the underlying `StyledText` component.
+ * @returns {JSX.Element} The styled text element.
+ */
+export const Typography: React.FC<TypographyProps> = ({
+  variant = 'regular',
+  style,
+  ...rest
+}: TypographyProps): JSX.Element => {
+  const styles = typographyVariantStyles[variant]
```
Comment (ngochuyduong) [inline src/components/Typography/Typography.tsx:70]: Update the name to `const variantStyles = typographyVariantStyles[variant]` to distinguish between other styles, Because `styles, style, ... ` is quite similar, so not easy to know which is the variant style, which is the text style at the first look unless we go to the styles definition.

Just a tiny suggestion. it's up to you
